### DISCUSSION TITLE
Executing commands to Pods or Containers

### DISCUSSION THREAD

**Discussion by koureasstavros** [kubernetes-client/csharp#1425] [Q&A] [answered]
Is there any method to execute commands from C# Kubernetes Client Programmatically?

For example I need to execute this command inside the pod/container:
`/venv/bin/python /pgadmin4/setup.py --load-servers servers.json`

Or outside the pod like kubectl
`kubectl exec unit-set-pgadmin-0 --  /venv/bin/python /pgadmin4/setup.py --load-servers servers.json`

Any ideas?

**Comment by tg123** [comment, marked as answer]
check exec example 
https://github.com/kubernetes-client/csharp/tree/master/examples/exec

**Reply by koureasstavros** [reply]
Thanks for the reply, I have tried this kind of code before (ConnectPostNamespacedPodExec and WebSocketNamespacedPodExecAsync) and also I checked out this code example (which seems very nice) but I cannot make it work with more complicated commands.

For example using WebSocketNamespacedPodExecAsync we should specify the command as vargs, so I have tried the following:

```
//string[] command_load = { "/venv/bin/python", "/pgadmin4/setup.py", "--load-servers servers/servers.json", "--user user@domain.com" };
//string[] command_load = { "/venv/bin/python /pgadmin4/setup.py --load-servers servers/servers.json --user user@domain.com" };
var websocket_load = client.WebSocketNamespacedPodExecAsync(podname, space, command_load, container, true, true, false, false).ConfigureAwait(false)
```
But does not work (code continues normally) and not seems to work actually, any idea?

**Reply by koureasstavros** [reply]
I just made it to work by splitting the command in every space (vargs) and waiting a little bit so the async operation completes.
